core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -32,83 +30,48 @@
+ * 6 | VALUE_REF_OFFSET        | these 2 integers together, represented as long  | entry that
+ * --|                        | provide VALUE_REF_OFFSET. Pay attention that     | was allocated
+ * 7 |                        | VALUE_REF_OFFSET is different than KEY_REF_OFFSET | first
  * -----------------------------------------------------------------------------|
  * 8 | NEXT  - entry index of the entry following the entry with entry index 2  |
  * -----------------------------------------------------------------------------|
  * 9 | NOT IN USE:             saved for alignment and future usage             |
  * --------------------------------------------------------------------------------------
- * 10| KEY_REFERENCE          | these 2 integers together, represented as long  | entry with
- * --|                        | provide KEY_REFERENCE. Pay attention that       | entry index
- * 11|                        | KEY_REFERENCE is different than VALUE_REFERENCE | 2
+ * 10| KEY_REF_OFFSET          | these 2 integers together, represented as long  | entry with
+ * --|                        | provide KEY_REF_OFFSET. Pay attention that       | entry index
+ * 11|                        | KEY_REF_OFFSET is different than VALUE_REF_OFFSET | 2
  * -----------------------------------------------------------------------------|
  * ...
  *
  *
  * Internal class, package visibility
  */
-class EntrySet<K, V> {
+class EntrySet<K, V> extends EntryArray<K, V> {
 
     /*-------------- Constants --------------*/
 
     /***
-     * This enum is used to access the different fields in each entry.
-     * The value associated with each entry signifies the offset of the field relative to the entry's beginning.
+     * NEXT - the next index of this entry (one integer). Must be with offset 0, otherwise, copying an entire
+     * entry should be fixed (In function {@code copyPartNoKeys}, search for "LABEL").
      */
-    private enum OFFSET {
-        /***
-         * NEXT - the next index of this entry (one integer). Must be with offset 0, otherwise, copying an entire
-         * entry should be fixed (In function {@code copyPartNoKeys}, search for "LABEL").
-         *
-         * KEY_REFERENCE - the blockID, length and offset of the value pointed from this entry (size of two
-         * integers, one long).
-         *
-         * VALUE_REFERENCE - the blockID, offset and version of the value pointed from this entry (size of two
-         * integers, one long). Equals to INVALID_REFERENCE if no value is point.
-         */
-        NEXT(0), KEY_REFERENCE(1), VALUE_REFERENCE(2);
-
-        final int value;
-
-        OFFSET(int value) {
-            this.value = value;
-        }
-    }
-
-    static final int INVALID_ENTRY_INDEX = 0;
-
-    // location of the first (head) node - just a next pointer (always same value 0)
-    private final int headNextIndex = 0;
+    private static final int NEXT_FIELD = 2;
```

Comment:
Rename to offset

Reply:
I think that "entry field" describes it better.
Anyway, the comment is not correct anymore.

Reply:
This number is used as an offset to next field inside an entry